modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/portlet/action/AddObjectFieldKeyCandidatesMVCResourceCommand.java

```diff
@@ -0,0 +1,134 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.object.web.internal.object.definitions.portlet.action;
+
+import com.liferay.object.constants.ObjectPortletKeys;
+import com.liferay.object.model.ObjectDefinition;
+import com.liferay.object.model.ObjectField;
+import com.liferay.object.service.ObjectDefinitionLocalService;
+import com.liferay.object.service.ObjectEntryLocalService;
+import com.liferay.object.service.ObjectFieldLocalService;
+import com.liferay.petra.sql.dsl.Column;
+import com.liferay.petra.sql.dsl.Table;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.json.JSONUtil;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.portlet.JSONPortletResponseUtil;
+import com.liferay.portal.kernel.portlet.bridges.mvc.BaseMVCResourceCommand;
+import com.liferay.portal.kernel.portlet.bridges.mvc.MVCResourceCommand;
+import com.liferay.portal.kernel.service.UserLocalService;
+import com.liferay.portal.kernel.servlet.SessionErrors;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.portlet.ResourceRequest;
+import javax.portlet.ResourceResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Pedro Leite
+ */
+@Component(
+	property = {
+		"javax.portlet.name=" + ObjectPortletKeys.OBJECT_DEFINITIONS,
+		"mvc.command.name=/object_definitions/add_object_field_key_candidates"
+	},
+	service = MVCResourceCommand.class
+)
+public class AddObjectFieldKeyCandidatesMVCResourceCommand
+	extends BaseMVCResourceCommand {
+
+	@Override
+	protected void doServeResource(
+			ResourceRequest resourceRequest, ResourceResponse resourceResponse)
+		throws Exception {
+
+		long objectDefinitionId = ParamUtil.getLong(
+			resourceRequest, "objectDefinitionId");
+		long[] objectFieldsIds = ParamUtil.getLongValues(
+			resourceRequest, "objectFieldsIds");
+
+		List<String> objectFieldLabels = new ArrayList<>();
+
+		ObjectDefinition objectDefinition =
+			_objectDefinitionLocalService.fetchObjectDefinition(
+				objectDefinitionId);
+
+		User user = _userLocalService.getUser(objectDefinition.getUserId());
+
+		for (Long objectFieldsId : objectFieldsIds) {
+			ObjectField objectField = _objectFieldLocalService.fetchObjectField(
+				objectFieldsId);
```

Comment:
If the source format doesn't complain I think you could inline like this:
```suggestion
		List<String> objectFieldLabels = new ArrayList<>();

		ObjectDefinition objectDefinition =
			_objectDefinitionLocalService.fetchObjectDefinition(
				objectDefinitionId);

		User user = _userLocalService.getUser(objectDefinition.getUserId());
			for (Long objectFieldsId :
				ParamUtil.getLongValues(resourceRequest, "objectFieldsIds")) {
			ObjectField objectField = _objectFieldLocalService.fetchObjectField(
				objectFieldsId);
```